godot-core/src/bind.rs

```diff
@@ -24,6 +24,66 @@ use crate::obj::GodotClass;
 ///
 /// Do not call any of these methods directly -- they are an interface to Godot. Functionality
 /// described here is available through other means (e.g. `init` via `Gd::new_default`).
+///
+/// It is not enough to impl GodotExt to be registered in Godot, for this you should look at [ExtensionLibrary](crate::init::ExtensionLibrary).
+///
+/// # Examples
+///
+/// ## Example with RefCounted as a base
+///
+/// ```
+/// use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// struct MyRef;
+///
+/// #[godot_api]
+/// impl MyRef {
+/// #[func]
+/// pub fn hello_world(&mut self) {
+///      godot_print!("Hello World!")
+///    }
+/// }
+///
+/// #[godot_api]
+/// impl GodotExt for MyRef {
+///     fn init(_: Base<RefCounted>) -> Self {
+///         MyRef
+///     }
+/// }
+/// ```
+///
+/// The following example allows to reference MyStruct in GDScript by calling for instance
+/// `MyStruct.new().hello_world()`.
+///
+///
+/// Note that you have to implement init otherwise you won't be able to call new or any
+/// other methods from GDScript.
+///
+/// ## Example with Node as a Base
+///
+/// ```
+/// use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(base=Node)]
+/// pub struct MyNode {
+///     #[base]
+///     base: Base<Node>,
+/// }
+///
+///
+/// #[godot_api]
+/// impl GodotExt for MyNode {
+///     fn init(base : Base<Node>) -> Self {
+///         MyNode {base}
+///     }
```

Comment:
Maybe run the examples through rustfmt:
```suggestion
///     fn init(base: Base<Node>) -> Self {
///         MyNode { base }
///     }
```

Reply:
it probably doesn't format doc test i assume ...

Reply:
there's an easy workaround, you remove /// from each line of your doctests and let cargo fmt do the work